trace/client/src/main/scala/org/http4s/otel4s/middleware/trace/client/ClientMiddleware.scala

```diff
@@ -30,41 +29,29 @@ import fs2.Stream
 import org.http4s.client.Client
 import org.typelevel.otel4s.trace.SpanKind
 import org.typelevel.otel4s.trace.StatusCode
+import org.typelevel.otel4s.trace.Tracer
 import org.typelevel.otel4s.trace.TracerProvider
 
-/** Middleware builder for wrapping an http4s `Client` to add tracing.
+/** Middleware for wrapping [[org.http4s.client.Client HTTP `Client`s]] to add
+  * tracing.
   *
   * @see [[https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-client]]
   */
-class ClientMiddlewareBuilder[F[_]: TracerProvider: Concurrent] private (
-    spanDataProvider: SpanDataProvider,
-    perRequestTracingFilter: PerRequestTracingFilter,
-) {
-  private[this] def copy(
-      perRequestTracingFilter: PerRequestTracingFilter
-  ): ClientMiddlewareBuilder[F] =
-    new ClientMiddlewareBuilder[F](
-      this.spanDataProvider,
-      perRequestTracingFilter,
-    )
+sealed trait ClientMiddleware[F[_]] {
 
-  /** Sets a filter that determines whether each request and its response
-    * should be traced.
+  /** @return a wrapped [[org.http4s.client.Client `Client`]] with added
+    *          tracing
     */
-  def withPerRequestTracingFilter(
-      perRequestTracingFilter: PerRequestTracingFilter
-  ): ClientMiddlewareBuilder[F] =
-    copy(perRequestTracingFilter = perRequestTracingFilter)
+  def wrap(client: Client[F]): Client[F]
+}
 
-  /** @return the configured middleware */
-  def build: F[Client[F] => Client[F]] =
```

Comment:
the motivation for this PR was the observation that `ClientMiddlewareBuilder#build` took no parameters and returned `F[Client[F] => Client[F]]`, while the three `build<HttpType>` methods on `ServerMiddleware` all took the `HttpType` as a parameter and returned `F[HttpType]`, which was inconsistent. I decided that `ClientMiddlewareBuilder` had the better and more flexible approach, though they might as well return named types with docs rather than simple functions.